test/spec/provider/cloud-element-templates/properties/CustomProperties.json

```diff
@@ -255,10 +274,197 @@
         "label": "Pattern (Integer)",
         "description": "Must be integer",
         "type": "String",
+        "group": "input",
+        "binding": {
+          "type": "property",
+          "name": "name"
+        },
+        "constraints": {
+          "pattern": {
+            "message": "Must be positive integer",
+            "value": "\\d+"
+          }
+        }
+      },
+      {
+        "label": "TextArea - NotEmpty",
+        "description": "Must not be empty",
+        "type": "Text",
+        "group": "textarea",
+        "binding": {
+          "type": "property",
+          "name": "name"
+        },
+        "constraints": {
+          "notEmpty": true
+        }
+      },
+      {
+        "label": "TextArea - MinLength",
+        "description": "Must have min length 5",
+        "type": "Text",
+        "group": "textarea",
+        "binding": {
+          "type": "property",
+          "name": "name"
+        },
+        "constraints": {
+          "minLength": 5
+        }
+      },
+      {
+        "label": "TextArea - MaxLength",
+        "description": "Must have max length 5",
+        "type": "Text",
+        "group": "textarea",
         "binding": {
           "type": "property",
           "name": "name"
         },
+        "constraints": {
+          "maxLength": 5
+        }
+      },
+      {
+        "label": "TextArea - Pattern (String)",
+        "description": "Must match /A+B/",
+        "type": "Text",
+        "group": "textarea",
+        "binding": {
+          "type": "property",
+          "name": "name"
+        },
+        "constraints": {
+          "pattern": "A+B"
+        }
+      },
+      {
+        "label": "TextArea - Pattern (String + Message)",
+        "description": "Must be https url",
+        "type": "Text",
+        "group": "textarea",
+        "binding": {
+          "type": "property",
+          "name": "name"
+        },
+        "constraints": {
+          "pattern": {
+            "message": "Must start with https://",
+            "value": "https://.*"
+          }
+        }
+      },
+      {
+        "label": "TextArea - Pattern (Integer)",
+        "description": "Must be integer",
+        "type": "Text",
+        "group": "textarea",
+        "binding": {
+          "type": "property",
+          "name": "name"
+        },
+        "constraints": {
+          "pattern": {
+            "message": "Must be positive integer",
+            "value": "\\d+"
+          }
+        }
+      },
+      {
```

Comment:
I don't see where we validate this. Could you help me out?

Reply:
Ah, it is https://github.com/bpmn-io/bpmn-js-properties-panel/pull/922/files#diff-b45460173a6452b61501a862ae9797960a113256f389211874da60d2f98ee020R1239.